src/main/scala/Main.scala

```diff
@@ -53,14 +53,19 @@ class SharedAxisCharts extends VBox {
 
     val lastChart = charts.last.chart
 
-    if (event.isShiftDown) {  // TODO implement vertical zoom/pan
+    val (newYLower, newYUpper) = if (event.isShiftDown) {
       val increment = -event.getDeltaX  // shifts X/Y axes: https://stackoverflow.com/questions/42429591/javafx-shiftscrollwheel-always-return-0-0
-
-    } else {
-
+      val range = lastChart.yUpper.value - lastChart.yLower.value
+      val mouseFrac = 1 - event.getY / lastChart.getHeight
+      val mouseTime = lastChart.yLower.value + (range * mouseFrac)
```

Comment:
style: it's no longer time here, it's the y-value the mouse is over, perhaps rename `mouseValue`?